src/chat/UnreadNotice.js

```diff
@@ -50,22 +50,24 @@ class UnreadNotice extends PureComponent<Props> {
 
   render() {
     const { limited, narrow, unreadCount } = this.props;
-
+    const shouldShow = unreadCount > 0;
     return (
-      <AnimatedScaleComponent visible={unreadCount > 0}>
-        <View style={styles.unreadContainer}>
-          <View style={styles.unreadTextWrapper}>
-            <RawLabel
-              style={styles.unreadText}
-              text={limited ? unreadToLimitedCount(unreadCount) : unreadCount.toString()}
-            />
-            <Label
-              style={styles.unreadText}
-              text={unreadCount === 1 ? 'unread message' : 'unread messages'}
-            />
+      <AnimatedScaleComponent visible={shouldShow}>
+        {shouldShow && (
```

Comment:
Why do we have this check inside every `AnimatedScaleComponent` children? We should move it to `AnimatedScaleComponent.js` instead.